### DISCUSSION TITLE
Inclusion Keeps Failing (Leviton ZW15R)

### DISCUSSION THREAD

**Discussion by mwaterbu** [zwave-js/zwave-js#6896] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
I am trying to get a Leviton ZW15R to include (using official Z-Wave JS HA add-on 0.5.0), but am running into issues. The first 20 or so times I tried, the device would time out of inclusion with no indication from ZWaveJS that any device was found. I thought it might be a signal strength/interference issue, even though my HUSBZB is just on the other side of the wall, so I moved the device to be right next to the controller.
(As a quick sidenote about the signal and distance from the hub: the same room also has multiple other ZWave devices, including a device which is a repeater, and those have no connection issues. It seems like the inclusion process isn’t using the repeaters to help it find new devices.)

After a couple more tries (factory resets of device, reboots of controller, etc), it gave me a message that inclusion failed (without first showing any inclusion progress), and to check the log for details. (Log attached below)

I don’t tend to have too many ZWave inclusion issues, I have 20+ devices in my mesh that work great.

What did you expect to happen instead?
Device pairs properly.

1. Enter programming mode: hold button for 7 seconds (status light turns amber)
2. Release button (status light blinks)
3. Begin inclusion on controller
4. Tap button one time (status light quickly flashes green)
5. Controller either reports that no device was found (times out), or reports that device could not be included, and to check the logs

Anything else we should know?
I moved the device for pairing to be outside of an electrical box/wall, only a couple feet away from the ZWave stick (which is plugged in to a USB 2 port with an extension cable), with clear line of sight, to eliminate any possibility of signal interference.

### Software versions

Driver (node-zwave-js): 12.5.5

Home Assistant Z-Wave Integration: 2024.5.2

Home Assistant Z-Wave JS Addon: 0.5.0

### Device information

Manufacturer: Leviton
Model name: ZW15R
Node ID: N/A

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[ZWaveJS_debug_inclusion.log](https://github.com/user-attachments/files/15597763/ZWaveJS_debug_inclusion.log)

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
A failed inclusion often boils down to one of the following things, most of which you already tried:
- The device needs to be excluded or factory reset
- The device needs a restart
- The controller needs a restart
- The device and controller have troubles communicating.

Since I see 3 identical incoming meter reports from node 3 in those 7 seconds you logged, I think you may have a bit too much noise on your Z-Wave network, be it from external sources or too much unsolicited reporting. It's probably the latter, but I'd need to see longer logs to know.
Some things you can try (requires Z-Wave JS UI though to get to the bottom of it all):
- Check the background noise levels (Z-Wave JS UI network map, click on the controller node). All channels should be roughly the same value, around -100 to -110 for 700/800 series, around -85 to -95 for 500.
- Tone down unnecessary reporting: https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=optimizing-the-reporting-configuration
- Check the link quality to existing nodes: https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=testing-the-connection-strength

**Reply by mwaterbu** [reply]
I followed the tips for reducing unnecessary reporting, definitely had some of those parameters set to send timed reports too often. To eliminate any chance for those clogging up my controller though, I simply unplugged them all and tried inclusion again. It still failed in the same way. I will keep trying...

**Comment by zhazell** [comment]
I've had really hard times including my Leviton ZW15R outlets as well. I had to move my raspi literally next to it with the USB stick so it would pair successfully. Once paired it was okay, but would drop off very regularly and show up as "dead". Only a breaker reset would bring them back online for a little bit and most of the time it would only receive an "alive" signal and then trying to send anything to it would make it "dead" again.

I found out from Hubitat forums that there is a beta firmware for the ZW15R version v1.11.0 which is based on the z-wave SDK v7.15.4. I signed up for the beta by emailing them at dssupport@leviton.com requesting access to the beta firmware. They will send you a document to sign and the firmware file. So once you get it included, I had to move my raspi next to the outlet again and push the OTA firmware to the outlet. After the upgrade, they have been MUCH more reliable and have not dropped off yet. 

Mine came from the factory with firmware FW: v1.8.1 which is based off SDK: v7.12.2. Seems to be quite buggy on this version.

**Reply by mwaterbu** [reply]
Thanks for the info! If I ever get it to pair, I will try upgrading the firmware.
Interesting that it is still in beta after over a year, especially since the fix makes the device significantly better than its current state (an expensive paperweight).

**Reply by mwaterbu** [reply]
> I had to move my raspi literally next to it

This information was really helpful, thanks again. I finally got it to include, but only when the receptacle was _right_ next to my controller. I only figured it out after I read this again, and decided maybe my 3ft line of sight wasn't close enough. I ran some tests after including it, and it works fine if I leave everything where it was when I got it to pair, but as soon as I move it a little further away the node drops out. Basically, it works at 1.5ft away, but not 3ft away. I have never seen a device with such poor signal. Other devices are over 100ft away through multiple walls/floors, and have no issues.

I also have FW 1.8.1, and have requested the beta, hope I can get that soon.
